kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/ErrorCollector.kt

```diff
@@ -2,49 +2,24 @@
 
 package io.kotest.assertions
 
-import java.util.Stack
-
-actual val errorCollector: ErrorCollector = ThreadLocalErrorCollector
-
-object ThreadLocalErrorCollector : ErrorCollector {
-
-   private val clueContext = object : ThreadLocal<Stack<Clue>>() {
-      override fun initialValue(): Stack<Clue> = Stack()
-   }
-
-   private val failures = object : ThreadLocal<MutableList<Throwable>>() {
-      override fun initialValue(): MutableList<Throwable> = mutableListOf()
-   }
-
-   private val collectionMode = object : ThreadLocal<ErrorCollectionMode>() {
-      override fun initialValue() = ErrorCollectionMode.Hard
-   }
-
-   override fun setCollectionMode(mode: ErrorCollectionMode) = collectionMode.set(mode)
-
-   override fun getCollectionMode(): ErrorCollectionMode = collectionMode.get()
-
-   override fun pushClue(clue: Clue) {
-      clueContext.get().push(clue)
+import kotlinx.coroutines.asContextElement
+
+actual val errorCollector: ErrorCollector get() = ThreadLocalErrorCollector.instance.get()
+
+/**
+ * A [CoroutineContext.Element] which keeps the error collector synchronized with thread-switching coroutines.
+ *
+ * When using [withClue] or [assertSoftly] on the JVM without the Kotest framework, this context element
+ * should be added to each top-level coroutine context, e.g. via
+ * - runBlocking(errorCollectorContextElement) { ... }
+ * - runBlockingTest(Dispatchers.IO + errorCollectorContextElement) { ... }
+ */
+val errorCollectorContextElement get() = ThreadLocalErrorCollector.instance.asContextElement()
```

Comment:
Does this need the get() if you're returning the same instance ?

Reply:
Conceptually, it is the same instance, that's why I made it a `val` here, too. The `get()` ensures that we don't have to worry about the internals of `asContextElement()`, which is invoked.